### PR TITLE
Re-work PullRequest reviewer request

### LINKED ISSUE #597: Add support for creating Review Requests
I just saw the update in #538 that added support for the GET methods for review requests, but it looks like the API for review requests is out of beta now, and the [POST methods are now available](https://developer.github.com/v3/pulls/review_requests/#create-a-review-request).  It would be awesome to be able to do these through the Python API.

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Add create_review_request() and delete_review_request() methods to
PullRequest, as well as renaming get_reviewer_requests() to
get_review_requests() and cleaning up its return value to firstly respect
teams, and secondly to cut out the middleman of PullRequestReviewerRequests,
which has been removed.

Fixes #597

### CODE REVIEW

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k Looks okay to me, can you please fix the test?

**Comment by s-t-e-v-e-n-k** [comment]
I forced pushed after travis failed, so I think the TravisBuddy comment is out of date.

File: github/PullRequest.py [hunk 1]
```diff
@@ -438,6 +437,46 @@ def create_review(self, commit, body, event=github.GithubObject.NotSet, comments
         self._useAttributes(data)
         return github.PullRequestReview.PullRequestReview(self._requester, headers, data, completed=True)
 
+    def create_review_request(self, reviewers=github.GithubObject.NotSet, team_reviewers=github.GithubObject.NotSet):
+        """
+        :calls `POST /repos/:owner/:repo/pulls/:number/requested_reviewers <https://developer.github.com/v3/pulls/review_requests/>`_
+        :param reviewers: iterable of strings
```
Comment (sfdye) [inline github/PullRequest.py]: Maybe for consistency, just `list` of strings?

File: github/PullRequest.py [hunk 2]
```diff
@@ -438,6 +437,46 @@ def create_review(self, commit, body, event=github.GithubObject.NotSet, comments
...
+        headers, data = self._requester.requestJsonAndCheck(
+            "POST",
+            self.url + "/requested_reviewers",
+            input=post_parameters
+        )
+
+    def delete_review_request(self, reviewers=github.GithubObject.NotSet, team_reviewers=github.GithubObject.NotSet):
+        """
+        :calls `DELETE /repos/:owner/:repo/pulls/:number/requested_reviewers <https://developer.github.com/v3/pulls/review_requests/>`_
+        :param reviewers: iterable of strings
```
Comment (sfdye) [inline github/PullRequest.py]: Here too

**Comment by sfdye** [comment]
Renamed the title just to remind me there are breaking changes when I prepare for the change log.

**Comment by s-t-e-v-e-n-k** [comment]
I've correct the class name in the title, I'll force push in a moment for iterable -> list.

**Pull request merged**